src/context.rs

```diff
@@ -75,3 +75,37 @@ impl Context {
         (self.hostname.as_str(), self.http_port).to_socket_addrs()
     }
 }
+
+#[test]
+#[allow(unused_variables)]
+fn test_should_add_a_service() {
+    use service::{ Service, ServiceProperties };
+    use iron::{Request, Response, IronResult};
+
+    struct ServiceStub;
+
+    impl Service for ServiceStub {
+        fn get_properties(&self) -> ServiceProperties {
+            ServiceProperties {
+                id: '1'.to_string(),
+                name: "dummy service".to_string(),
+                description: "really nothing to see".to_string(),
+                http_url: '2'.to_string(),
+                ws_url: '3'.to_string()
+            }
+        }
+        fn start(&self)  {}
+        fn stop(&self) {}
+        fn process_request(&self, req: &Request) -> IronResult<Response> {
+            return Ok(Response::new());
+        }
+    }
+
+    let hostname: Option<String> = Some("localhost".to_string());
```

Comment:
use to_owned() instead of to_string(), and inline it in Context::new(false, ...)